src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -45,17 +47,19 @@ class GraphQLTest extends Specification {
                 .field(
                 newFieldDefinition()
                         .name("id")
+                        .dataFetcher(new MapDataFetcher("id"))
```

Comment:
> Is this interchangeable with a FieldDataFetcher?

Nope. [`MapDataFetcher`](https://github.com/dminkovsky/graphql-java/blob/eb75d57f0bbcaed455812218fd81beda028a8379/src/main/java/graphql/schema/MapDataFetcher.java#L17-L21) takes behavior that used to be part of both `PropertyDataFetcher` and `FieldDataFetcher`. Currently, both of these fetchers attempt to use the source as a `Map` before doing anything else. In this PR, neither of these fetchers attempt using the source as a `Map`.